include/dali/core/geom/box.h

```diff
@@ -102,17 +105,32 @@ constexpr DALI_HOST_DEV CoordinateType volume(const Box<ndims, CoordinateType> &
   return dali::volume(box.extent());
 }
 
-
 /**
  * @return Intersection of two boxes or a default one when the arguments are disjoint.
  */
-template<int ndims, typename CoordinateType>
-constexpr DALI_HOST_DEV Box<ndims, CoordinateType>
-intersection(const Box<ndims, CoordinateType> &lhs, const Box<ndims, CoordinateType> &rhs) {
+template <int ndims, typename CoordinateType>
+constexpr DALI_HOST_DEV Box<ndims, CoordinateType> intersection(
+    const Box<ndims, CoordinateType> &lhs, const Box<ndims, CoordinateType> &rhs) {
   Box<ndims, CoordinateType> tmp = {max(lhs.lo, rhs.lo), min(lhs.hi, rhs.hi)};
   return any_coord(tmp.hi <= tmp.lo) ? Box<ndims, CoordinateType>() : tmp;
```

Comment:
actually, this function has wrong NaN handling - should be:
```suggestion
  return !all_coords(tmp.hi > tmp.lo) ? Box<ndims, CoordinateType>() : tmp;
```